### PR TITLE
fix:UI validation for existing email during user registration

### PR DESCRIPTION
**Pull request by Anxhul10**
## Proposed changes (including videos or screenshots)

Screenshot from 2025-02-20 22-41-08

[Screencast from 2025-02-20 22-40-53.webm](https://github.com/user-attachments/assets/92f02d64-8848-4ca1-bd37-c4ca20310758)

## Issue(s)

#35263

### CODE REVIEW

**Comment by Anxhul10** [comment]
I just realized that 'tasked' issues are already assigned. 
As tasked issues  have been referred to the devs for them to fix. Should I close this PR, or would it still be helpful in any way? I'll make sure to pick unassigned issues next time.

**Comment by Anxhul10** [comment]
> > I just realized that 'tasked' issues are already assigned. As tasked issues have been referred to the devs for them to fix. Should I close this PR, or would it still be helpful in any way? I'll make sure to pick unassigned issues next time.
> 
> I don't think anyone is working on that at the moment, so it should be fine
> 
> Can you fix the lint errors?
> 
> Thanks!

I have fixed the lint errors .
but it fails at  CI / deploy-preview (pull_request) Failing after 3m

**Review by MartinSchoeler — APPROVED** [review]

File: packages/web-ui-registration/src/RegisterForm.tsx [hunk 1]
```diff
@@ -93,7 +93,7 @@ export const RegisterForm = ({ setLoginRoute }: { setLoginRoute: DispatchLoginRo
 					if (error.errorType === 'error-user-already-exists') {
 						setError('username', { type: 'user-already-exists', message: t('registration.component.form.usernameAlreadyExists') });
 					}
-					if (/Email already exists/.test(error.error)) {
+					if (/Email is already in use/.test(error.error)) {
```
Comment (KevLehman) [inline packages/web-ui-registration/src/RegisterForm.tsx:96]: Can't we use the same error `Email already exists` so we don't change this one?

File: apps/meteor/app/api/server/v1/users.ts [hunk 2]
```diff
@@ -649,7 +650,9 @@ API.v1.addRoute(
 			if (!(await checkUsernameAvailability(this.bodyParams.username))) {
 				return API.v1.failure('Username is already in use');
 			}
-
+			if (!(await checkEmailAvailability(this.bodyParams.email))) {
+				return API.v1.failure('Email is already in use');
```
Comment (KevLehman) [inline apps/meteor/app/api/server/v1/users.ts:654]:
```suggestion
				return API.v1.failure('Email already exists');
```

File: packages/web-ui-registration/src/RegisterForm.tsx (hunk 1 above)
Comment (KevLehman) [inline packages/web-ui-registration/src/RegisterForm.tsx:96]:
```suggestion
					if (/Email already exists/.test(error.error)) {
```

**Review by KevLehman — CHANGES_REQUESTED** [review]

**Comment by julio-rocketchat** [comment]
Security-wise, as long as we’re able to guarantee that this endpoint is protected by rate-limiting, I would say this is fine. I don’t see a problem with showing that an email has already been registered in this context, as long as we prevent massive user enumeration. We can proceed with it.

**Review by MarcosSpessatto — CHANGES_REQUESTED** [review]
Can we have tests covering the scenario?

**Comment by Anxhul10** [comment]
Just to confirm, do you want me to run tests in my fork before making the PR?

**Comment by jessicaschelly** [comment]
> Just to confirm, do you want me to run tests in my fork before making the PR? or I have to follow this https://developer.rocket.chat/docs/assist-with-testing-rocketchat

We are requesting automated end-to-end test cases for the scenario: _validate existing email during user registration_

Please ensure the test covers expected behavior and edge cases (if applicable). You can refer to the E2E testing guidelines here: [Rocket.Chat E2E Testing](https://github.com/RocketChat/Rocket.Chat/wiki/E2E-Testing).

Feel free to let us know if you need any more clarification.

**Comment by Anxhul10** [comment]
> > Just to confirm, do you want me to run tests in my fork before making the PR? or I have to follow this https://developer.rocket.chat/docs/assist-with-testing-rocketchat
> 
> We are requesting automated end-to-end test cases for the scenario: _validate existing email during user registration_
> 
> Please ensure the test covers expected behavior and edge cases (if applicable). You can refer to the E2E testing guidelines here: [Rocket.Chat E2E Testing](https://github.com/RocketChat/Rocket.Chat/wiki/E2E-Testing).
> 
> Feel free to let us know if you need any more clarification.

Added test for UI you can check the screen recording

https://github.com/user-attachments/assets/8c04e8d0-45c4-4828-b84f-d78a2e99b090

File: apps/meteor/tests/e2e/register.spec.ts [hunk 3]
```diff
@@ -134,6 +134,39 @@ test.describe.parallel('register', () => {
 
 			expect(results.violations).toEqual([]);
 		});
+
+		test('should not allow registration with an already registered email', async ({ page }) => {
```
Comment (jessicaschelly) [inline apps/meteor/tests/e2e/register.spec.ts:138]: I noticed that these tests are inside the `Registration form Disabled` 
We could have a new describe for these types of validation errors, such as `Registration form validation` wdyt?

Comment (Anxhul10) [reply apps/meteor/tests/e2e/register.spec.ts:138]: correct me if I am wrong ,the test 'should not allow registration with an already registered email' is under test 'Successfully Registration flow' but you are right that these kind of validation errors should be in different describe block probably "validation errors". wdyt?

[the remark this answers is not in the document]
Comment (Anxhul10) [reply apps/meteor/tests/e2e/register.spec.ts:145]: done !!

**Review by MarcosSpessatto — APPROVED** [review]

**Review by jessicaschelly — APPROVED** [review]

**Review by KevLehman — DISMISSED** [review]

**Review by dougfabris — APPROVED** [review]
